src/features/window-location/main/for-top-bar-state-handler.injectable.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getMessageChannelListenerInjectable } from "../../../common/utils/channel/message-channel-listener-injection-token";
+import getVisibleWindowsInjectable from "../../../main/start-main-application/lens-window/get-visible-windows.injectable";
+import topBarStateInjectable from "../../top-bar/common/state.injectable";
+import { windowLocationChangedChannel } from "../common/channel";
+
+const windowLocationHasChangedForTopBarStateHandlerInjectable = getMessageChannelListenerInjectable({
+  id: "for-top-bar-state",
+  channel: windowLocationChangedChannel,
+  handler: (di) => {
+    const getVisibleWindows = di.inject(getVisibleWindowsInjectable);
+    const topBarState = di.inject(topBarStateInjectable);
+
+    return () => {
+      const windows = getVisibleWindows();
+
+      topBarState.set({
+        prevEnabled: windows.some(window => window.canGoBack()),
+        nextEnabled: windows.some(window => window.canGoForward()),
+      });
```

Comment:
This seems fundamentally wrong. At the moment we have only one window, but if we had more, this would result in very weird behaviour.

Reply:
Fair enough, however it is what the current state was so I just changed it to be injectable.